File: examples/university-demo-OLD/src/main/java/io/axoniq/demo/university/faculty/events/CourseCreated.java
package io.axoniq.demo.university.faculty.events;

import io.axoniq.demo.university.faculty.FacultyTags;
import io.axoniq.demo.university.shared.ids.CourseId;
import io.axoniq.demo.university.shared.ids.FacultyId;
import org.axonframework.eventsourcing.annotations.EventTag;

public record CourseCreated(
        @EventTag(key = FacultyTags.FACULTY_ID)
        FacultyId facultyId,
        @EventTag(key = FacultyTags.COURSE_ID)
        CourseId courseId,
        String name,
        int capacity
) {

}

File: examples/university-demo-kotlin/src/main/kotlin/faculty/events/CourseCreated.kt
package org.axonframework.examples.university.faculty.events

import org.axonframework.eventsourcing.annotation.EventTag
import org.axonframework.examples.university.faculty.FacultyTags
import org.axonframework.examples.university.shared.ids.CourseId

data class CourseCreated(
    @EventTag(key = FacultyTags.COURSE)
    val courseId: CourseId,
    val name: String,
    val capacity: Int
) : FacultyEvent
